tests/thread/host/spinlock_test_host.cpp

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <errno.h>
+#include <openenclave/host.h>
+#include <openenclave/internal/error.h>
+#include <openenclave/internal/tests.h>
+#include <signal.h>
+#include <stdio.h>
+#include <cassert>
+#include <chrono>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <thread>
+#include "thread_u.h"
+
+#define ERRNO_TEST_TIMES 10
+#define LARGE_ENOUGH_MEMORY_SIZE 1024 * 1024 * 1024
+#define NUM_SPINLOCK_TEST_THREADS 2
+
+int g_flag_withlock = 0;
+static uint32_t g_lock = 0;
+static int g_error_num = 0;
+
+void* errno_thread1(void* para)
+{
+    int i = 0;
+    int t_errno = 0;
+    void* buffer = NULL;
+    oe_enclave_t* enclave = (oe_enclave_t*)para;
+
+    for (i = 0; i < ERRNO_TEST_TIMES; i++)
+    {
+        if (g_flag_withlock)
+        {
+            enc_spin_lock(enclave, &g_lock);
```

Comment:
We don't recommend that locks are beyond the duration of an ecall. The reason is that in Open Enclave SDK, the binding between host thread and the enclave thread lasts only for the duration of the ecall.
Another ecall from the same host thread may end up using a different enclave thread. 

As a result, we avoid holding enclave locks beyond the duration of the ecall.

Reply:
I would request the test be changed the following way: 
1. while calling into an enclave don’t acquire a spin lock. 
2. The errno is stored in an out arg in the ecall
3. On the host end after the ecall is complete, acquire a lock and manipulate the global var to store the errno you just received. This is not necessary and you can use two global vars as well.

Reply:
Thanks for your suggestion, i updated the code.